infobip-jackson-extension-api/src/main/java/com/infobip/jackson/PresentPropertyJsonTypeResolver.java

```diff
@@ -14,11 +14,33 @@ public PresentPropertyJsonTypeResolver(Class<E> type) {
     @Override
     public final Class<?> resolve(Map<String, Object> json) {
         for (E constant : type.getEnumConstants()) {
-            if (json.containsKey(constant.toString().toLowerCase())) {
+            if (json.containsKey(resolvePropertyName(constant))) {
```

Comment:
It's now evident that default use case of camelCase field to camelCase json property doesn't work (it works only for simple names with which there's no casing actually).
I suggest you add guava(29.0-jre) to this project and introduce CaseFormat field to this class which should be overridable by a specific constructor and for current one the default should be CaseFormat.LOWER_CAMEL.
Then instead of new resolvePropertyName(constant) and related new infrastructure `CaseFormat.UPPER_UNDERSCORE.to(caseFormat, constant.toString())` on this line should be sufficient.

Reply:
Ok, I'll try with this approach 👍

Reply:
Fixed as you suggested. Tests as well as readme tuned to reflect current changes.